go-controller/pkg/util/kube.go

```diff
@@ -69,7 +71,17 @@ func newKubernetesRestConfig(conf *config.KubernetesConfig) (*rest.Config, error
 			Host:        conf.APIServer,
 			BearerToken: conf.Token,
 		}
-		if conf.CACert != "" {
+		if len(conf.CAData) > 0 {
+			caData, err := base64.StdEncoding.DecodeString(conf.CAData)
+			if err != nil {
+				return nil, fmt.Errorf("kubernetes CA base64 data invalid: %v", err)
+			}
+
+			if _, err := cert.NewPoolFromBytes(caData); err != nil {
+				return nil, err
+			}
+			kconfig.TLSClientConfig = rest.TLSClientConfig{CAData: caData}
+		} else if conf.CACert != "" {
```

Comment:
Why not just deprecate writing to a file and always pass Ca data?

Reply:
@trozet do you mean always just read in the ca file to the data, and never pass the file path around internally?

Reply:
Right

Reply:
It Shall Be Done

Reply:
It Is Done